File: src/main/java/com/example/balloon/model/dto/game/GameConfig.java
package com.example.balloon.model.dto.game;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.*;

@Data
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class GameConfig {
    private int gameDuration;
    private int maxScore;
    private int minPlayTime;
    private String rewardName;
    private int bonusEvery;
    private int bonusScore;
    private double rewardChance;
    private double bonusRewardChance;
    private boolean enableRewards;
    private boolean enableBonuses;
    private boolean allowReplay;
    private int maxGamesPerDay;
    private double scoreMultiplier;

    public static GameConfig defaults() {
        GameConfig cfg = new GameConfig();
        cfg.setGameDuration(30);
        cfg.setMaxScore(0);
        cfg.setMinPlayTime(5);
        cfg.setRewardName("Супер награда");
        cfg.setBonusEvery(4);
        cfg.setBonusScore(500);
        cfg.setRewardChance(1);
        cfg.setBonusRewardChance(1);
        cfg.setEnableRewards(true);
        cfg.setEnableBonuses(true);
        cfg.setAllowReplay(true);
        cfg.setMaxGamesPerDay(0);
        cfg.setScoreMultiplier(1);
        return cfg;
    }
}

File: internal/service/game_config.go
package service

import (
	"Otbor/internal/database"
	"Otbor/internal/models"
	"encoding/json"
	"errors"

	"github.com/redis/go-redis/v9"
)

const gameConfigRedisKey = "game:config"

// DefaultGameConfig возвращает конфиг "по умолчанию" — он воспроизводит
// поведение, которое раньше было зашито в код напрямую (хардкод),
// чтобы до первой настройки через админку игра работала как прежде.
func DefaultGameConfig() models.Config {
	return models.Config{
		GameDuration:      30,
		MaxScore:          0, // 0 = без ограничения
		MinPlayTime:       5,
		RewardName:        "Супер награда",
		BonusEvery:        4,
		BonusScore:        500,
		RewardChance:      1, // 100% — награда выдается всегда, как раньше
		BonusRewardChance: 1,
		EnableRewards:     true,
		EnableBonuses:     true,
		AllowReplay:       true,
		MaxGamesPerDay:    0, // 0 = без ограничения
		ScoreMultiplier:   1,
	}
}

// GetGameConfig читает текущий конфиг игры из Redis. Если конфиг еще
// ни разу не сохранялся, возвращает DefaultGameConfig().
func GetGameConfig() (models.Config, error) {
	data, err := database.RDB.Get(database.Ctx, gameConfigRedisKey).Result()

	if errors.Is(err, redis.Nil) {
		return DefaultGameConfig(), nil
	}

	if err != nil {
		return models.Config{}, err
	}

	// Отталкиваемся от дефолтов, чтобы поля, не сохраненные в старой
	// версии конфига (при добавлении новых полей в будущем), не
	// схлопывались в нулевые значения.
	cfg := DefaultGameConfig()

	if err := json.Unmarshal([]byte(data), &cfg); err != nil {
		return models.Config{}, err
	}

	return cfg, nil
}

// SetGameConfig сохраняет конфиг игры в Redis целиком.
func SetGameConfig(cfg models.Config) error {
	data, err := json.Marshal(cfg)
	if err != nil {
		return err
	}

	return database.RDB.Set(
		database.Ctx,
		gameConfigRedisKey,
		data,
		0,
	).Err()
}
